app/assets/javascripts/admin/modules/breadcrumb_preview.js

```diff
@@ -0,0 +1,66 @@
+(function(Modules) {
+  "use strict";
+
+  Modules.BreadcrumbPreview = function() {
+    var that = this;
+
+    that.fetchCurrentCheckboxes = function() {
+      return $('.topic-tree :checked');
+    };
+
+    that.buildBreadcrumbsStructure = function(checkboxes) {
+      var structure = $.map(checkboxes, function(checkbox) {
+        var $checkbox = $(checkbox);
+        var ancestors = $checkbox.data('ancestors').split('|');
+
+        return [ancestors];
+      });
+
+      return structure;
+    };
+
+    that.filterBreadcrumbs = function(breadcrumbs) {
+      var longestFirst = breadcrumbs.sort(function (a, b) {
+        return b.length - a.length;
+      });
+
+      var seenAlready = [];
+
+      $.each(longestFirst, function(index, breadcrumb) {
+        var seen = false;
+
+        $.each(seenAlready, function(index, seenBreadcrumb) {
+          var breadcrumbString = JSON.stringify(breadcrumb);
```

Comment:
Might be worth making a helper method (or importing one from [lodash](https://lodash.com/docs/4.17.4#isEqual)) so from the code it's obvious what you're trying to do.

Either that or could add a comment.